apple/testing/default_runner/ios_test_runner.template.sh

```diff
@@ -146,10 +158,62 @@ else
   )
 fi
 
+exit_status=0
 cmd=("%(testrunner_binary)s"
   "${runner_flags[@]}"
   "${target_flags[@]}"
   "${custom_xctestrunner_args[@]}")
-"${cmd[@]}" 2>&1
-status=$?
-exit ${status}
+"${cmd[@]}" 2>&1 || exit_status=$?
+
+if [[ "$exit_status" -ne 0 ]]; then
+  exit "$exit_status"
+fi
+
+if [[ "${COVERAGE:-}" -ne 1 ]]; then
+  # Normal tests run without coverage
+  exit 0
+fi
+
+readonly profdata="$TMP_DIR/coverage.profdata"
+xcrun llvm-profdata merge "$profraw" --output "$profdata"
+
+# llvm-cov export doesn't print any warnings/errors when files that it's trying
+# to reference don't exist. Unfortunately these are absolute paths that won't
+# be valid across machines. When using llvm-cov show it prints a warning for
+# this so we can validate we don't hit this case.
+error_file=$(mktemp)
+llvm_cov_status=0
+xcrun llvm-cov \
+  show \
+  -instr-profile "$profdata" \
+  -path-equivalence="$ROOT","$PWD" \
```

Comment:
this is the biggest part I'm unsure about, this isn't right for folks using the remote cache, they need to use coverage-prefix-map and remap `.` to `$PWD`

Reply:
I'll leave this and we can fix if something becomes more clear